chanzuckerberg/idseq-web
# Show host subtracted in pipeline info sidebar

# Description

In preparation for user-entered hosts that default to ERCC-only, this adds some info to disambiguate cases where the host subtracted is not the same as the host entered. 

![image](https://user-images.githubusercontent.com/28797/73024115-c88f6c80-3de1-11ea-9686-462c5f12a634.png)

# Notes

* On upload, users will be warned about no subtraction if they enter an unknown host
* The `host_subtracted` method depends on internal structure of dag json (similar to pipeline viz) and s3 URLs. To be safe, it always returns `nil` in case of missing data, and no row displayed in the sidebar.

# Tests

* new rspec test
* load sample, see info in sidebar
* load ERCC only sample, see custom string

![image](https://user-images.githubusercontent.com/28797/73024467-7ef35180-3de2-11ea-966d-04102a309dd9.png)
